### DISCUSSION TITLE
Easiest PC monitor backlight setup using Plasma Stick 2040 W and SK6812 LEDs

### DISCUSSION THREAD

**Discussion by kliszaj** [awawa-dev/HyperHDR#739] [Q&A] [answered]
I'm new to all of this so apologies if I'm asking a stupid question.  I recently set up HyperHDR running on an RPi Zero W 2 to a single strip of SK6812 LEDs using the SK6812spi driver (data line connect to GPIO 10 in RPi).  From my computer, I'm running [Hyperion Screen Capture](https://sabaatworld.github.io/HyperionScreenCap/) to wirelessly send frame data to the RPi which then controls the LEDs.  

I've read the helpful warning in HyperHDR that SK6812spi is not supported and a different controller should be used instead.  I just ordered a Plasma Stick 2040 W and I'm wondering how to best set this up so that my PC wirelessly sends frame data that can then be used to control the LEDs? Do I need to send frame data from my PC to the RPi running HyperHDR which is then connected to the Plasma Stick 2040 W? How should I connect the Plasma Stick 2040W to the RPi? Or is there a simpler way?

**Comment by awawa-dev** [comment, marked as answer]
Hi
No need for an external app anymore: HyperHDR v20 includes an updated DirectX11 grabber that uses a pixel and vertex shader for even faster screen capture, even supporting Windows HDR modes natively. I propose to connect a HyperHDR instance on Windows (no LEDs attached, just for screen capturing) to another HyperHDR instance on RPi (with Plasma Stick 2040W connected using USB, HyperHDR actually acts similar to WLED) using the UDP raw sender/receiver mechanism.

This makes synchronization even faster because we do not send entire images over the network (Hyperion Screen Capture), only the colors of the RGB LEDs, and therefore much smaller packets.

![obraz](https://github.com/awawa-dev/HyperHDR/assets/69086569/03836ce7-0bd8-40cd-9076-eb09c694eb6f)

**Reply by kliszaj** [reply]
Very helpful! Thank you. And then do I just connect GP15 of the Plasma Stick 2040W to the data line of the LED?

**Reply by awawa-dev** [reply]
If you uploaded proper firmware for Plasma Stick 2040W https://github.com/awawa-dev/HyperSerialPico
then connect LED data line to 'PIXELS' output at the end (it's actually GPIO15). You must also connect Plasma ground ('-') to the LED strip ground to make a common ground with the LED power supply ground. Do not connect anything to Plasma Stick 2040W '5v' pin.

**Reply by awawa-dev** [reply]
I've updated the wiki with a step-by-step sync setup.
https://github.com/awawa-dev/HyperHDR/wiki/Quick-start#synchronization-between-different-hyperhdr-devices
If something is still unclear, please ask.

**Reply by kliszaj** [reply]
Thanks for the tips - it all seems to be working flawlessly.  One more question - do you recommend to do color channel adjustment on the sender side or the receiver side?

**Reply by awawa-dev** [reply]
Good question... if you use RPi on a daily basis, disable 'Image processing' on Windows. For example for the red/green/blue gamma settings, change 1.5 to 1.
Otherwise, gamma will be calculated twice: once on Windows and then again on RPi, which will significantly reduce the LED brightness.

**Reply by kliszaj** [reply]
Do you think it would be better to connect the PC that is running HyperHDR directly to the Plasma Stick 2040W rather than communicating wirelessly through an RPi (is that even possible)?  I'm noticing some minor latency with the lights and wondering how to best eliminate it.

**Reply by awawa-dev** [reply]
Before you do this, increase the capture rate in the DirectX grabber settings (eg. default 10 to 60FPS). If this doesn't help, you can also turn off smoothing on both instances.

**Reply by kliszaj** [reply]
That plus adjusting the smoothing on the receiving side really did the trick.
